app/ui-react/syndesis/src/modules/integrations/components/editor/SelectConnectionPage.tsx

```diff
@@ -135,27 +139,38 @@ export class SelectConnectionPage extends React.Component<
   public render() {
     return (
       <WithRouteData<ISelectConnectionRouteParams, ISelectConnectionRouteState>>
-        {(params, state, { history }) => {
+        {(params, state) => {
           const { flowId, position } = params;
           const { integration = getEmptyIntegration() } = state;
           const positionAsNumber = parseInt(position, 10) || 0;
-          const onStepClick = (connectionOrStep: ConnectionOverview | Step) => {
-            const stepKind = getStepKind(connectionOrStep);
-            switch (stepKind) {
+          const getStepHref = (step: IUIStep) => {
+            switch (step.uiStepKind) {
               case 'api-provider':
-                history.push(this.props.apiProviderHref(params, state));
-                break;
+                return this.props.apiProviderHref(step, params, state);
               case 'endpoint':
-                history.push(
-                  this.props.connectionHref(
-                    connectionOrStep as ConnectionOverview,
-                    {
-                      ...params,
-                    },
-                    state
-                  )
+              case 'connector':
+                return this.props.connectionHref(
+                  step as ConnectionOverview,
+                  params,
+                  state
                 );
-                break;
+              case 'expressionFilter':
+              case 'ruleFilter':
+                return this.props.filterHref(step, params, state);
+              case 'extension':
+                return this.props.extensionHref(step, params, state);
+              case 'mapper':
+                return this.props.mapperHref(step, params, state);
+              case 'headers':
+                throw new Error(`Can't handle stepKind ${step.stepKind}`);
```

Comment:
@gashcrumb do you have any idea about this stepKind?

Reply:
Hmm, that's a new one.  I wonder if it's one of the extensions there on staging.  Maybe we need a default case that just sends those to the regular step configuration page that just uses autoform.